cmd/zetatool/filterdeposit/evm.go

```diff
@@ -0,0 +1,208 @@
+		fmt.Printf("adding segment: %d, startblock: %d\n", segment, startBlock)
+		deposits = append(deposits, GetHashListSegment(client, startBlock, endBlock, cfg)...)
+		startBlock = endBlock
+		endBlock = endBlock + cfg.EvmMaxRange
+		if endBlock > latestBlock {
+			endBlock = latestBlock
+		}
+		segment++
+	}
+	return deposits
+}
+
+// GetHashListSegment queries and filters deposits for a given range
+func GetHashListSegment(client *ethclient.Client, startBlock uint64, endBlock uint64, cfg *config.Config) []Deposit {
+	deposits := make([]Deposit, 0)
+
+	connectorAddress := common.HexToAddress(cfg.ConnectorAddress)
+	connectorContract, err := zetaconnector.NewZetaConnectorNonEth(connectorAddress, client)
+	if err != nil {
+		fmt.Println("error: ", err.Error())
+	}
+	erc20CustodyAddress := common.HexToAddress(cfg.CustodyAddress)
+	erc20CustodyContract, err := erc20custody.NewERC20Custody(erc20CustodyAddress, client)
+	if err != nil {
+		fmt.Println("error: ", err.Error())
+	}
+
+	custodyIter, err := erc20CustodyContract.FilterDeposited(&bind.FilterOpts{
+		Start:   startBlock,
+		End:     &endBlock,
+		Context: context.TODO(),
+	}, []common.Address{})
+	if err != nil {
+		fmt.Println("error loading filter: ", err.Error())
+		return deposits
+	}
+
+	connectorIter, err := connectorContract.FilterZetaSent(&bind.FilterOpts{
+		Start:   startBlock,
+		End:     &endBlock,
+		Context: context.TODO(),
+	}, []common.Address{}, []*big.Int{})
+	if err != nil {
+		fmt.Println("error loading filter: ", err.Error())
+		return deposits
+	}
+
+	// ********************** Get ERC20 Custody Deposit events
+	for custodyIter.Next() {
+		// sanity check tx event
+		err := CheckEvmTxLog(&custodyIter.Event.Raw, erc20CustodyAddress, "", TopicsDeposited)
+		if err == nil {
+			//fmt.Println("adding deposits")
+			deposits = append(deposits, Deposit{
+				TxID:   custodyIter.Event.Raw.TxHash.Hex(),
+				Amount: custodyIter.Event.Amount.Uint64(),
+			})
+		}
+	}
+
+	// ********************** Get Connector ZetaSent events
+	for connectorIter.Next() {
+		// sanity check tx event
+		err := CheckEvmTxLog(&connectorIter.Event.Raw, connectorAddress, "", TopicsZetaSent)
+		if err == nil {
+			//fmt.Println("adding deposits")
+			deposits = append(deposits, Deposit{
+				TxID:   connectorIter.Event.Raw.TxHash.Hex(),
+				Amount: connectorIter.Event.ZetaValueAndGas.Uint64(),
+			})
+		}
+	}
+
+	//********************** Get Transactions sent directly to TSS address
+	tssDeposits, err := getTSSDeposits(cfg.TssAddressEVM, startBlock, endBlock)
+	if err != nil {
+		fmt.Printf("getTSSDeposits returned err: %s", err.Error())
+	}
+	deposits = append(deposits, tssDeposits...)
+
+	return deposits
+}
+
+// getTSSDeposits more specifically queries and filters deposits based on direct transfers the TSS address.
+func getTSSDeposits(tssAddress string, startBlock uint64, endBlock uint64) ([]Deposit, error) {
+	client := etherscan.New(etherscan.Mainnet, "S3AVTNXDJQZQQUVXJM4XVIPBRYECGK88VX")
```

Comment:
`S3AVTNXDJQZQQUVXJM4XVIPBRYECGK88VX`
What this represent?

Reply:
That is the etherscan API key, I removed it from the repo. I must have forgotten it there when testing.